src/lib/list/list.ts

```diff
@@ -46,35 +46,30 @@ export class MdListDivider {}
 /** A Material Design list component. */
 @Component({
   moduleId: module.id,
-  selector: 'md-list, mat-list, md-nav-list, mat-nav-list',
-  host: {'role': 'list'},
+  selector: 'md-nav-list, mat-nav-list',
+  host: {
+    'role': 'navigation',
+    'class': 'mat-nav-list'
+  },
   template: '<ng-content></ng-content>',
   styleUrls: ['list.css'],
   inputs: ['disableRipple'],
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class MdList extends _MdListMixinBase implements CanDisableRipple {}
+export class MdNavList extends _MdListMixinBase implements CanDisableRipple {}
 
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
+@Component({
+  moduleId: module.id,
   selector: 'md-list, mat-list',
-  host: {'class': 'mat-list'}
-})
-export class MdListCssMatStyler {}
-
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
-  selector: 'md-nav-list, mat-nav-list',
-  host: {'class': 'mat-nav-list'}
+  template: '<ng-content></ng-content>',
```

Comment:
I think that in general we still keep the templates in a separate HTML file, even if it's a one-liner.